x/crosschain/keeper/msg_server_add_to_outtx_tracker.go

```diff
@@ -29,6 +29,19 @@ func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToO
 		return nil, observertypes.ErrSupportedChains
 	}
 
+	// the cctx must exist and be pending
+	cctx, err := k.CctxByNonce(ctx, &types.QueryGetCctxByNonceRequest{
+		ChainID: msg.ChainId,
+		Nonce:   msg.Nonce,
+	})
+	if err != nil || cctx == nil || cctx.CrossChainTx == nil {
+		return nil, cosmoserrors.Wrap(types.ErrCannotFindCctx, "cannot add out tx: no corresponding cctx found")
```

Comment:
We should return a specific error depending on the condition that got met

Reply:
@lumtis Did you mean creating another error (instead of `ErrCannotFindCctx `) or adding detailed info to the err message?

Reply:
Showing a different error if
err != nil and return the error
cctx == nil
cctx. CrossChainTx and return the cctx data

Reply:
the [commit](https://github.com/zeta-chain/node/pull/1533/commits/04271a208d8d300f24be371ab8b1710c24abe1af)